File: ch04/switchbtn/switchbtn_ex3.c
#include <stdio.h>
#include <pigpiod_if2.h> 
#define PINNO 23
int count;
void cb_func(int pi, unsigned user_gpio, unsigned level, uint32_t tick)
{
    printf("count : %d\n", ++count);
    printf("pi:%d user_gpio:%d level:%d tick:%u\n\n", pi, user_gpio, level, tick);
}

int main()
{
    int pi;
    if((pi = pigpio_start(NULL, NULL)) < 0){
        fprintf(stderr, "%s\n", pigpio_error(pi));
        return 1;
    }

    set_mode(pi, PINNO, PI_INPUT);
    set_pull_up_down(pi, PINNO, PI_PUD_UP);

    // callback function 설정
    callback(pi, PINNO, FALLING_EDGE, cb_func);

    getchar();

    pigpio_stop(pi);
    return 0;
}

File: ch04/switchbtn/switchbtn_ex3.py
#!/usr/bin/python3
import pigpio
import time

count = 0
def cbf(gpio, level, tick):
    global count
    count += 1
    print("count:", count)
    print("gpio:", gpio, " level:", level, " tick:", tick, "\n")

GPIO = 23

pi = pigpio.pi()
if not pi.connected:
    exit()

pi.set_mode(GPIO, pigpio.INPUT)
pi.set_pull_up_down(GPIO, pigpio.PUD_UP)

pi.callback(GPIO, pigpio.FALLING_EDGE, cbf)
input()

pi.stop()
